fbw-common/src/systems/shared/src/navigraph/client.ts

```diff
@@ -0,0 +1,345 @@
+                        topRight: {
+                            lat: chart.planview.bbox_geo[3],
+                            lon: chart.planview.bbox_geo[2],
+                            xPx: chart.planview.bbox_local[2],
+                            yPx: chart.planview.bbox_local[3],
+                        },
+                        width: chart.bbox_local[2],
+                        height: chart.bbox_local[1],
+                    } : undefined,
+                }));
+
+                return {
+                    arrival: chartArray.filter((chart) => chart.type.category === 'ARRIVAL'),
+                    approach: chartArray.filter((chart) => chart.type.category === 'APPROACH'),
+                    airport: chartArray.filter((chart) => chart.type.category === 'AIRPORT'),
+                    departure: chartArray.filter((chart) => chart.type.category === 'DEPARTURE'),
+                    reference: chartArray.filter((chart) => (
+                        (chart.type.category !== 'ARRIVAL')
+                        && (chart.type.category !== 'APPROACH')
+                        && (chart.type.category !== 'AIRPORT')
+                        && (chart.type.category !== 'DEPARTURE')
+                    )),
+                };
+            }
+        }
+
+        return emptyNavigraphCharts;
+    }
+
+    public async getAirportInfo(icao: string): Promise<AirportInfo | null> {
+        if (this.hasToken) {
+            const chartJsonUrl = await this.chartCall(icao, 'airport.json');
+
+            const chartJsonResp = await fetch(chartJsonUrl);
+
+            if (chartJsonResp.ok) {
+                const chartJson = await chartJsonResp.json();
+
+                return { name: chartJson.name };
+            }
+        }
+
+        return null;
+    }
+
+    public get hasToken(): boolean {
+        return !!this.accessToken;
+    }
+
+    public async assignUserName(): Promise<void> {
+        if (this.hasToken) {
+            try {
+                const userInfoResp = await fetch('https://identity.api.navigraph.com/connect/userinfo', { headers: { Authorization: `Bearer ${this.accessToken}` } });
+
+                if (userInfoResp.ok) {
+                    const userInfoJson = await userInfoResp.json();
+
+                    this.userName = userInfoJson.preferred_username;
+                    NXDataStore.set('NAVIGRAPH_USERNAME', this.userName);
+                }
+            } catch (_) {
+                console.log('Unable to Fetch User Info. #NV103');
+            }
+        }
+    }
+
+    public async subscriptionStatus(): Promise<NavigraphSubscriptionStatus> {
+        if (this.hasToken) {
+            const decodedToken = JSON.parse(atob(this.accessToken.split('.')[1]));
+
+            const subscriptionTypes = decodedToken.subscriptions as string[];
+
+            if (subscriptionTypes.includes('fmsdata') && subscriptionTypes.includes('charts')) {
+                return NavigraphSubscriptionStatus.Unlimited;
```

Comment:
This would probably be better as flags. Navi might change their subscription offerings in future.

Reply:
We have to do code changes regardless if they change the subscription offerings, so I think it's moot